### DISCUSSION TITLE
XS-A14 RGB Smart Plug - LEDLink Emulation

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#17603] [Q&A]
I am running Tasmota on following Device

[https://templates.blakadder.com/XS-A14.html](https://templates.blakadder.com/XS-A14.html)

![image](https://user-images.githubusercontent.com/84637540/210719790-d9f3134f-cdb7-4d59-aee9-600fbea2eeb7.png)

I accidentally discovered that with following configuration, during WiFi connection establishment Relay gets toggled continously until WiFi connection is established. On further investigation it was discovered that during connection establishment the Blue LED (GPIO12) gets a pulse.

![image](https://user-images.githubusercontent.com/84637540/210722716-562f1ae8-30e4-4f7e-9606-2d1d9035d63b.png)

I configured GPIO12 as LEDLink and confirmed that it gets triggered during WiFi Establishment.

Is there way to configure GPIO12 with PWM and still get LinkLED like behavior during WiFi Establishment (perhaps a RULE)?

Can the PIN Assignment be changed via RULE (or script)?

**Comment by Jason2866** [comment]
Looks like relay is wrongly assigned to a PWM led output. Try to find the real GPIO pin.

**Reply by jkwim** [reply]
The configuration shown on the LEFT is not correct. Configuration on the RIGHT is the correct assignment (Relay +Button are working fine with this) .

What I am talking about is that when you have the config on the left, the Relay gets toggled during WiFi connection establishment suggesting that the PIN works as LEDLink. However to make the RGB work this pin has to be defined as PWM. When it is PWM the WiFi indicator function does not work as it is not a direct LED.

I am exploring the possibility of making the pin working as a LED until WiFi gets connected and thereafter to work as PWM.
